lib/Parse/ParseExpr.cpp

```diff
@@ -1914,6 +1914,10 @@ ParserResult<Expr> Parser::parseExprPrimary(Diag<> ID, bool isExprBasic) {
   default:
   UnknownCharacter:
     checkForInputIncomplete();
+    if (Context.LangOpts.hasFeature(Feature::TrailingComma) &&
+        Tok.is(tok::eof) && Tok.getText() == ")") {
```

Comment:
Could you check for `Tok.is(tok::r_paren)` instead of comparing the text?

Reply:
So... I don't know why but `Tok.is(tok::r_paren)` doesn't work here 🫠

Reply:
Do you know what the token’s kind is? Otherwise I’ll try and start debugging it because I’m curious now.

Reply:
Also curious, that if we `checkForInputComplete` wouldn't a call to `!isInputIncomplete()` achieve the same as this check `Tok.is(tok::eof) && Tok.getText() == ")"` does? (note condition is different so that is why I think maybe negate it could have same result)